be-java-springboot/templates/gradle-repositories.template

```diff
@@ -0,0 +1,16 @@
+    if (!no_nexus) {
+        def nexusMaven = { repoUrl ->
+            maven {
+                credentials {
+                    username = "${nexus_user}"
+                    password = "${nexus_pw}"
+                }
+                url repoUrl
+            }
+        }
+        nexusMaven("${nexus_url}/repository/jcenter/")
+        nexusMaven("${nexus_url}/repository/maven-public/")
+        nexusMaven("${nexus_url}/repository/atlassian_public/")
+    }
+    mavenCentral()
```

Comment:
@renedupont `mavenCentral()` and `jcenter` should be active if `no_nexus==true`, right? If yes, the `else` block is missing

Reply:
If `no_nexus = true` the if block is skipped and the first repo that gradle would contact is mavenCentral and then jcenter.
If `no_nexus = false` gradle would first look into our nexus and if the artifacts can't be found there (or nexus is not reachable for example) it would also check mavenCentral and jcenter. I see it more as a feature 😆

Reply:
I guess, I'm missing something here... the implementation above will always execute `m̀avenCentral()` and `jcenter()`, correct? Shouldn't be either nexusMaven or mavenCentral/jcenter based on `no_nexus`?

Reply:
As far as I understood [how gradle deals with dependencies](https://docs.gradle.org/current/userguide/dependency_resolution.html#obtaining_module_metadata) it goes by order until it finds a repo where it can download a dependency and then takes that repo for all others. That way it does not hurt to have publicly available repos AFTER our nexus repos.

Reply:
ok... but isn't the idea of using nexus to not depend on third party directly.
As I see it, the thing is either we use nexus or not, but not a mix. Do you see my point?

Reply:
not quite, nexus downloads from public repos anyway if it does not have a requested artifact. Therefore, currently gradle reaching out to mavencentral or jcenter directly is just a theoretical case that could happen when nexus for whatever unlikely reason cannot provide a certain artifact. But I don't have any hard feelings about this 'else', so I'll just add it 😆

Reply:
thanks! :-)